Library/Homebrew/dev-cmd/pr-upload.rb

```diff
@@ -30,29 +30,59 @@ def pr_upload_args
     end
   end
 
-  def check_bottled_formulae(json_files)
-    hashes = json_files.reduce({}) do |hash, json|
-      hash.deep_merge(JSON.parse(IO.read(json)))
+  def pr_upload
+    args = pr_upload_args.parse
+
+    json_files = Dir["*.json"]
+    odie "No JSON files found in the current working directory" if json_files.empty?
+
+    # Changes later to `true` if bottle's root_url matches.
+    github_releases = false
+
+    bottles_hash = json_files.reduce({}) do |hash, json_file|
+      hash.deep_merge(JSON.parse(IO.read(json_file)))
     end
 
-    hashes.each do |name, hash|
-      formula_path = HOMEBREW_REPOSITORY/hash["formula"]["path"]
+    bottles_hash.each do |name, bottle_hash|
+      formula_path = HOMEBREW_REPOSITORY/bottle_hash["formula"]["path"]
       formula_version = Formulary.factory(formula_path).pkg_version
-      bottle_version = PkgVersion.parse hash["formula"]["pkg_version"]
-      next if formula_version == bottle_version
+      bottle_version = PkgVersion.parse bottle_hash["formula"]["pkg_version"]
 
-      odie "Bottles are for #{name} #{bottle_version} but formula is version #{formula_version}!"
-    end
-  end
+      # Check if bottle version matches formula version.
+      if formula_version != bottle_version
+        odie "Bottles are for #{name} #{bottle_version} but formula is version #{formula_version}!"
+      end
 
-  def pr_upload
-    args = pr_upload_args.parse
+      # Check if bottle's root_url matches GitHub Releases url.
+      root_url = bottle_hash["bottle"]["root_url"]
+      url_match = root_url.match HOMEBREW_RELEASES_URL_REGEX
+      _, user, repo, tag = *url_match
 
-    bintray_org = args.bintray_org || "homebrew"
-    bintray = Bintray.new(org: bintray_org)
+      next unless tag
 
-    json_files = Dir["*.json"]
-    odie "No JSON files found in the current working directory" if json_files.empty?
+      # Define that we are uploading to GitHub Releases.
+      # Used later to return early, so we don't deal with Bintray.
+      github_releases = true
+
+      next if args.dry_run?
+
+      # Ensure a release is created.
+      release = begin
+        GitHub.get_release user, repo, tag
+        odebug "Existing GitHub release \"#{tag}\" found"
+      rescue GitHub::HTTPNotFoundError
+        odebug "Creating new GitHub release \"#{tag}\""
+        GitHub.create_or_update_release user, repo, tag
+      end
+
+      # Upload bottles as release assets.
+      bottle_hash["bottle"]["tags"].each do |_, tag_hash|
+        remote_file = tag_hash["filename"]
+        local_file = tag_hash["local_filename"]
+        odebug "Uploading #{remote_file}"
+        GitHub.upload_release_asset user, repo, release["id"], local_file: local_file, remote_file: remote_file
```

Comment:
This feels like it should probably happen around the same time we handle uploading to Bintray below. Thoughts?

Reply:
It could happen later, but that means duplicated code I'm afraid.

Reply:
Which code would need duplicated and why? Just trying to understand, sorry :)

Reply:
I think it should look better now :)